src/main/java/de/zalando/aruha/nakadi/service/CursorsService.java

```diff
@@ -112,16 +120,45 @@ private void createSubscriptionInZkIfNeeded(final Subscription subscription) thr
 
                             subscriptionClient.fillEmptySubscription(subscriptionOffsets);
                         }
-                    } catch (Exception e) {
+                    } catch (final Exception e) {
                         atomicReference.set(e);
                     }
                 });
             }
-        } catch (Exception e) {
+        } catch (final Exception e) {
             atomicReference.set(e);
         }
         if (atomicReference.get() != null) {
-            throw new ServiceUnavailableException("Error communicating with zookeeper", atomicReference.get());
+            throw new ServiceUnavailableException(ERROR_COMMUNICATING_WITH_ZOOKEEPER, atomicReference.get());
+        }
+    }
+
+    public List<Cursor> getSubscriptionCursors(final String subscriptionId) throws NakadiException {
+        final Subscription subscription = subscriptionRepository.getSubscription(subscriptionId);
+        final String eventTypeName = subscription.getEventTypes().iterator().next();
+        final String topic = eventTypeRepository.findByName(eventTypeName).getTopic();
+        final String partitionsPath = format(PATH_ZK_PARTITIONS, subscriptionId, topic);
+
+        try {
+            return zkHolder.get().getChildren().forPath(partitionsPath).stream()
+                    .map(partition -> readCursor(subscriptionId, topic, partition))
+                    .collect(Collectors.toList());
+        } catch (final KeeperException.NoNodeException nne) {
+            LOG.debug(nne.getMessage(), nne);
+            return Collections.emptyList();
+        } catch (final Exception e) {
+            LOG.error(e.getMessage(), e);
+            throw new ServiceUnavailableException(ERROR_COMMUNICATING_WITH_ZOOKEEPER, e);
+        }
+    }
+
+    private Cursor readCursor(final String subscriptionId, final String topic, final String partition) throws RuntimeException {
+        try {
+            final String offsetPath = format(PATH_ZK_OFFSET, subscriptionId, topic, partition);
+            final String currentOffset = new String(zkHolder.get().getData().forPath(offsetPath), CHARSET_UTF8);
+            return new Cursor(partition, currentOffset);
+        } catch (final Exception e) {
```

Comment:
It is thrown by curator framework. We checked for the path in the previous zk call and In case exception thrown here we can not do anything specific just failing with service unavailable because something wrong with zk state.